lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -361,16 +361,8 @@ public function getListTableIndexesSQL($table, $currentDatabase = null)
      */
     private function getTableWhereClause($table, $classAlias = 'c', $namespaceAlias = 'n')
     {
-        $whereClause = $namespaceAlias.".nspname NOT IN ('pg_catalog', 'information_schema', 'pg_toast') AND ";
-        if (strpos($table, ".") !== false) {
-            list($schema, $table) = explode(".", $table);
-            $schema = "'" . $schema . "'";
-        } else {
-            $schema = "ANY(string_to_array((select replace(replace(setting,'\"\$user\"',user),' ','') from pg_catalog.pg_settings where name = 'search_path'),','))";
-        }
-
-        $table = new Identifier($table);
-        $whereClause .= "$classAlias.relname = '" . $table->getName() . "' AND $namespaceAlias.nspname = $schema";
+        $tableNameEscaped = str_replace("'", "''", $table);
+        $whereClause = $classAlias.".oid='".$tableNameEscaped."'::regclass::oid";
```

Comment:
Can you clear me up on this? What does it exactly do? Does it work for full qualified table names? How does it know which namespace to look on for unqualified table names?